diff --git a/config.h.in b/config.h.in
index 794e00c..c470dfc 100644
--- a/config.h.in
+++ b/config.h.in
@@ -1,5 +1,8 @@
 /* config.h.in (GLPK configuration template file) */
 
+#undef __WOE__
+/* defined if building for MS Windows */
+
 #undef HAVE_SYS_TIME_H
 /* defined if the <sys/time.h> header can be used */
 
diff --git a/configure.ac b/configure.ac
index cbdb724..752e0b9 100644
--- a/configure.ac
+++ b/configure.ac
@@ -114,6 +114,12 @@ case $host_os in
       LIBODBC="libodbc.dylib"
       LIBMYSQL="libmysqlclient.dylib"
       ;;
+   cygwin* | mingw*)
+      LIBIODBC="odbc32.dll"
+      LIBODBC="odbc32.dll"
+      LIBMYSQL="libmysql.dll"
+      AC_DEFINE([__WOE__], [1], [N/A])
+      ;;
    *)
       LIBIODBC="libiodbc.so"
       LIBODBC="libodbc.so"
@@ -145,7 +151,6 @@ if test "$enable_mysql" = "yes"; then
       AC_MSG_ERROR([--enable-mysql requires --enable-dl])
    fi
    AC_MSG_RESULT([yes])
-   CPPFLAGS="-I/usr/include/mysql $CPPFLAGS"
    AC_DEFINE_UNQUOTED([MYSQL_DLNAME], ["$LIBMYSQL"], [N/A])
 else
    AC_MSG_RESULT([no])
