diff --git a/CMakeLists.txt b/CMakeLists.txt
index 426bd06..9e22369 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -415,62 +415,8 @@ if(SENTRY_WITH_LIBUNWINDSTACK)
 endif()
 
 if(SENTRY_BACKEND_CRASHPAD)
-	if(SENTRY_BUILD_SHARED_LIBS)
-		set(CRASHPAD_ENABLE_INSTALL OFF CACHE BOOL "Enable crashpad installation" FORCE)
-	else()
-		set(CRASHPAD_ENABLE_INSTALL ON CACHE BOOL "Enable crashpad installation" FORCE)
-	endif()
-	add_subdirectory(external/crashpad crashpad_build)
-
-	if(WIN32)
-		add_dependencies(sentry crashpad::wer)
-	endif()
-
-	# set static runtime if enabled
-	if(SENTRY_BUILD_RUNTIMESTATIC AND MSVC)
-		set_property(TARGET crashpad_client PROPERTY MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>")
-		set_property(TARGET crashpad_compat PROPERTY MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>")
-		set_property(TARGET crashpad_getopt PROPERTY MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>")
-		set_property(TARGET crashpad_handler PROPERTY MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>")
-		set_property(TARGET crashpad_handler_lib PROPERTY MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>")
-		set_property(TARGET crashpad_minidump PROPERTY MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>")
-		set_property(TARGET crashpad_snapshot PROPERTY MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>")
-		set_property(TARGET crashpad_tools PROPERTY MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>")
-		set_property(TARGET crashpad_util PROPERTY MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>")
-		set_property(TARGET crashpad_wer PROPERTY MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>")
-		set_property(TARGET crashpad_zlib PROPERTY MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>")
-		set_property(TARGET mini_chromium PROPERTY MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>")
-	endif()
-
-	if(DEFINED SENTRY_FOLDER)
-		set_target_properties(crashpad_client PROPERTIES FOLDER ${SENTRY_FOLDER})
-		set_target_properties(crashpad_compat PROPERTIES FOLDER ${SENTRY_FOLDER})
-		set_target_properties(crashpad_getopt PROPERTIES FOLDER ${SENTRY_FOLDER})
-		set_target_properties(crashpad_handler PROPERTIES FOLDER ${SENTRY_FOLDER})
-		set_target_properties(crashpad_handler_lib PROPERTIES FOLDER ${SENTRY_FOLDER})
-		set_target_properties(crashpad_minidump PROPERTIES FOLDER ${SENTRY_FOLDER})
-		set_target_properties(crashpad_snapshot PROPERTIES FOLDER ${SENTRY_FOLDER})
-		set_target_properties(crashpad_tools PROPERTIES FOLDER ${SENTRY_FOLDER})
-		set_target_properties(crashpad_util PROPERTIES FOLDER ${SENTRY_FOLDER})
-		set_target_properties(crashpad_zlib PROPERTIES FOLDER ${SENTRY_FOLDER})
-		set_target_properties(mini_chromium PROPERTIES FOLDER ${SENTRY_FOLDER})
-		set_target_properties(crashpad_wer PROPERTIES FOLDER ${SENTRY_FOLDER})
-	endif()
-
-	target_link_libraries(sentry PRIVATE
-		$<BUILD_INTERFACE:crashpad::client>
-		$<INSTALL_INTERFACE:sentry_crashpad::client>
-	)
-	install(EXPORT crashpad_export NAMESPACE sentry_crashpad:: FILE sentry_crashpad-targets.cmake
-		DESTINATION "${CMAKE_INSTALL_CMAKEDIR}"
-	)
-	if(WIN32 AND MSVC)
-		sentry_install(FILES $<TARGET_PDB_FILE:crashpad_handler>
-			DESTINATION "${CMAKE_INSTALL_BINDIR}" OPTIONAL)
-		sentry_install(FILES $<TARGET_PDB_FILE:crashpad_wer>
-			DESTINATION "${CMAKE_INSTALL_BINDIR}" OPTIONAL)
-	endif()
-	add_dependencies(sentry crashpad::handler)
+	find_package(crashpad CONFIG REQUIRED)
+	target_link_libraries(sentry PUBLIC crashpad::crashpad)
 elseif(SENTRY_BACKEND_BREAKPAD)
 	option(SENTRY_BREAKPAD_SYSTEM "Use system breakpad" OFF)
 	if(SENTRY_BREAKPAD_SYSTEM)
diff --git a/sentry-config.cmake.in b/sentry-config.cmake.in
index ac75b8d..a3ba6bf 100644
--- a/sentry-config.cmake.in
+++ b/sentry-config.cmake.in
@@ -13,6 +13,9 @@ if(NOT SENTRY_BUILD_SHARED_LIBS)
     if(SENTRY_TRANSPORT_COMPRESSION OR CRASHPAD_ZLIB_SYSTEM)
     	find_dependency(ZLIB)
     endif()
+    if(SENTRY_BACKEND STREQUAL "crashpad")
+        find_dependency(crashpad CONFIG)
+    endif()
     if(SENTRY_BACKEND STREQUAL "breakpad" AND SENTRY_BREAKPAD_SYSTEM)
     	find_dependency(PkgConfig)
     	pkg_check_modules(BREAKPAD REQUIRED IMPORTED_TARGET breakpad-client)
diff --git a/src/backends/sentry_backend_crashpad.cpp b/src/backends/sentry_backend_crashpad.cpp
index 6941f95..1a1822f 100644
--- a/src/backends/sentry_backend_crashpad.cpp
+++ b/src/backends/sentry_backend_crashpad.cpp
@@ -120,7 +120,7 @@ crashpad_backend_user_consent_changed(sentry_backend_t *backend)
     data->db->GetSettings()->SetUploadsEnabled(!sentry__should_skip_upload());
 }
 
-#ifdef SENTRY_PLATFORM_WINDOWS
+#if defined(SENTRY_PLATFORM_WINDOWS) && defined(SENTRY_TRANSPORT_CRASHPAD_USE_WER)
 static void
 crashpad_register_wer_module(
     const sentry_path_t *absolute_handler_path, const crashpad_state_t *data)
@@ -452,7 +452,7 @@ crashpad_backend_startup(
         return 1;
     }
 
-#ifdef SENTRY_PLATFORM_WINDOWS
+#if defined(SENTRY_PLATFORM_WINDOWS) && defined(SENTRY_TRANSPORT_CRASHPAD_USE_WER)
     crashpad_register_wer_module(absolute_handler_path, data);
 #endif
 
