diff --git a/include/sleepy_discord/json_wrapper.h b/include/sleepy_discord/json_wrapper.h
index b5fef8a..aefb67d 100644
--- a/include/sleepy_discord/json_wrapper.h
+++ b/include/sleepy_discord/json_wrapper.h
@@ -1,4 +1,5 @@
 #pragma once
+#include <algorithm>
 #include <list>
 #include <utility>
 #include <vector>
diff --git a/include/sleepy_discord/zlib_compression.h b/include/sleepy_discord/zlib_compression.h
index 43e6575..56df382 100644
--- a/include/sleepy_discord/zlib_compression.h
+++ b/include/sleepy_discord/zlib_compression.h
@@ -9,6 +9,7 @@
 #include <forward_list>
 #include <string>
 #include <mutex>
+#include <cstring>
 
 namespace SleepyDiscord {
 	//This Queue is basicly a single linked list with the back and size stored
